File: Ananya Chandra/Emirp.py
#program for  Emirp number

n=int(input("Enter the number\n"))
def prime(t,i):
    if t==1:
        return 0
    elif(i<t):
        if(t%i!=0):
            return prime(t,i=i+1)
        else:
            return 0
    else:
        return 1

p=prime(n,2)
if p==1:
    s=0
    c=n
    while(c>0):
        r=c%10
        s=s*10+r
        c=c//10
    if(s==n):
        p=0
    else:
        p=prime(s,2)
    if p==1:
        print(n," is an Emirp number")
    else:
        print(n, " is not an Emirp number")
else:
    print(n, " is not an Emirp number")

File: HarshKesarwani/emirp.cpp
#include <iostream>
using namespace std;

int backward(int n)
{
    int r = 0;
    for (int x = n; x != 0; x = x / 10)
    {
        int d = x % 10;
        r = r * 10 + d;
    }
    return r;
}

bool isPrime(int n)
{
    bool flag = true;
    for (int x = 2; x <= n / 2; x++)
    {
        if (n % x == 0)
        {
            flag = false;
            break;
        }
    }
    return flag;
}

int main()
{
    int num;
    cout << "Enter number to check if it is EMIRP number" << endl;
    cin >> num;
    int back_num = backward(num);
    if (isPrime(num) == true && isPrime(back_num) == true)
    {
        cout << num << " IS AN EMIRP NUMBER" << endl;
    }
    else
    {
        cout << num << " IS NOT AN EMIRP NUMBER" << endl;
    }
}
